### ISSUE TITLE
no-unmodified-loop-condition false positives for "set" functions and only 1 modified

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#16116]
### Environment

Node version: v16.14.2
npm version: v8.12.1
Local ESLint version: Not found
Global ESLint version: v8.17.0 (Currently used)
Operating System: linux 4.18.0-348.7.1.el8_5.x86_64

### What parser are you using?

Default (Espree)

### What did you do?

Simplified example, where start/end are `new Date` objects.

```js
while ( start <= end ) {
	start.setDate( start.getDate() + 1 );
}
```

### What did you expect to happen?

No error

start is modified by "setDate"
This function should be declared as "modifier", so it will not report an error for start.
Maybe even give an option to allow all "set" methods - if any "set" method is used on a variable, it will count as modified

end is not modified, but it's just used in the condition.
In my opinion no-unmodified-loop should only give error if both variables in a comparison condition are unchanged.

### What actually happened?

no-unmodified-loop-condition for start and end variables

**Comment by nzakas** [comment]
This is a limitation of the rule. “Modified” means that the variable was assigned a new value during the loop. ESLint doesn’t know that start and end are Date objects, we there’s no way to know that the value they represent is modified. You’ll need to manually disable the rule in this case using a disable comment.

**Comment by kkmuffme** [comment]
Could there perhaps be an option to allow certain functions, so we don't get that error? (e.g. `.set`)
